aroma/utils.py

```diff
@@ -10,247 +18,42 @@
-    the data has to be linearly registered to MNI152 (i.e. the user has a
-    reason not to use non-linear registration on the data).
+    The four features used for classification are: maximum RP correlation,
+    high-frequency content, edge-fraction, and CSF-fraction.
 
     Parameters
     ----------
-    fsl_dir : str
-        Full path of the bin-directory of FSL
-    in_file : str
-        Full path to the data file (nii.gz) which has to be registerd to
-        MNI152 T1 2mm
-    out_file : str
-        Full path of the output file
-    affmat : str
-        Full path of the mat file describing the linear registration (if data
-        is still in native space)
-    warp : str
-        Full path of the warp file describing the non-linear registration (if
-        data has not been registered to MNI152 space yet)
-
-    Output
-    ------
-    melodic_IC_mm_MNI2mm.nii.gz : merged file containing the mixture modeling
-                                  thresholded Z-statistical maps registered to
-                                  MNI152 2mm
-    """
-    # Define the MNI152 T1 2mm template
-    fslnobin = fsl_dir.rsplit("/", 2)[0]
-    ref = op.join(fslnobin, "data", "standard", "MNI152_T1_2mm_brain.nii.gz")
-
-    # If the no affmat- or warp-file has been specified, assume that the data
-    # is already in MNI152 space. In that case only check if resampling to
-    # 2mm is needed
-    if not affmat and not warp:
-        in_img = nib.load(in_file)
-        # Get 3D voxel size
-        pixdim1, pixdim2, pixdim3 = in_img.header.get_zooms()[:3]
-
-        # If voxel size is not 2mm isotropic, resample the data, otherwise
-        # copy the file
-        if (pixdim1 != 2) or (pixdim2 != 2) or (pixdim3 != 2):
-            os.system(
-                " ".join(
-                    [
-                        op.join(fsl_dir, "flirt"),
-                        " -ref " + ref,
-                        " -in " + in_file,
-                        " -out " + out_file,
-                        " -applyisoxfm 2 -interp trilinear",
-                    ]
-                )
-            )
-        else:
-            os.copyfile(in_file, out_file)
+    X : :obj:`pandas.DataFrame`
+        Features table (C x 4), must contain the following columns:
+        "edge_fract", "csf_fract", "max_RP_corr", and "HFC".
 
-    # If only a warp-file has been specified, assume that the data has already
-    # been registered to the structural scan. In that case apply the warping
-    # without a affmat
-    elif not affmat and warp:
-        # Apply warp
-        os.system(
-            " ".join(
-                [
-                    op.join(fsl_dir, "applywarp"),
-                    "--ref=" + ref,
-                    "--in=" + in_file,
-                    "--out=" + out_file,
-                    "--warp=" + warp,
-                    "--interp=trilinear",
-                ]
-            )
-        )
+    Returns
+    -------
+    Y : array_like
```

Comment:
A minor note, but since the predicted values should be a 1D array, I think `y` would be better.